tensorflow/core/kernels/dml_softmax_ops.cc

```diff
@@ -95,14 +118,43 @@ class DmlSoftmaxXentWithLogitsKernel : public DmlKernel {
     // The strides we need to set to broadcast class across an entire tensor
     dml::TensorDesc::Dimensions reduce_strides = {0, 1, 3};
 
+    // broadcast logits if the shape do not match
+    uint32_t toBCastBatchDim =
+        logits_shape.dim_size(0) == shape_in.dim_size(0) ? 0 : 1;
+    uint32_t toBCastClassDim =
+        logits_shape.dim_size(1) == shape_in.dim_size(1) ? 0 : 1;
+    dml::Expression logits_bcast;
+    if (toBCastBatchDim || toBCastClassDim) {
+      logits_bcast = dml::Reinterpret(
+          logits, input_sizes,
+          dml::TensorDesc::Dimensions(
+              {0, 0, 1 - toBCastBatchDim, 1 - toBCastClassDim}));
+    } else {
+      logits_bcast = dml::Reinterpret(logits, input_sizes, {});
+    }
+
+    // broadcast labels if the shape do not match
+    dml::Expression labels_bcast;
+    toBCastBatchDim = labels_shape.dim_size(0) == shape_in.dim_size(0) ? 0 : 1;
+    toBCastClassDim = labels_shape.dim_size(1) == shape_in.dim_size(1) ? 0 : 1;
+
+    if (toBCastBatchDim || toBCastClassDim) {
+      labels_bcast = dml::Reinterpret(
+          labels, input_sizes,
+          dml::TensorDesc::Dimensions(
+              {0, 0, 1 - toBCastBatchDim, 1 - toBCastClassDim}));
+    } else {
+      labels_bcast = dml::Reinterpret(labels, input_sizes, {});
+    }

```

Comment:
Same here.

```suggestion
    toBCastBatchDim = labels_shape.dim_size(0) == shape_in.dim_size(0) ? 0 : 1;
    toBCastClassDim = labels_shape.dim_size(1) == shape_in.dim_size(1) ? 0 : 1;

    if (toBCastBatchDim || toBCastClassDim) {
      labels = dml::Reinterpret(
          labels, input_sizes,
          dml::TensorDesc::Dimensions(
              {0, 0, 1 - toBCastBatchDim, 1 - toBCastClassDim}));
    }
``` #Resolved